frontend/src/views/FuelCodes/AddFuelCode/AddFuelCode.jsx

```diff
@@ -1,3 +1,57 @@
+import BCBox from '@/components/BCBox'
+import { columnDefs, defaultColDef } from './_schema'
+import { useRef, useState, useCallback } from 'react'
+import BCDataGridEditor from '@/components/BCDataGrid/BCDataGridEditor'
+import { Typography } from '@mui/material'
+
 export const AddFuelCode = () => {
-  return <div>AddFuelCode</div>
+  const [rowData, setRowData] = useState(null)
+  const [gridApi, setGridApi] = useState(null)
+  const [columnApi, setColumnApi] = useState(null)
+  const gridRef = useRef(null)
+
+  function onGridReady(params) {
+    setGridApi(params.api)
+    setColumnApi(params.columnApi)
+
+    fetch('http://localhost:4000/fuelCodes')
+      .then((res) => res.json())
+      .then((data) => {
+        setRowData(data.slice(0, 100))
+      })
+    params.api.sizeColumnsToFit()
+  }
+  const saveData = useCallback(() => {
+    const allRowData = []
+    gridApi.forEachNode((node) => allRowData.push(node.data))
+    const modifiedRows = allRowData.filter((row) => row.modified)
+    console.log(modifiedRows)
```

Comment:
This needs more work i take it?

Reply:
yes, we'll having a post api call to save the data.